src/main.c

```diff
@@ -1204,15 +1204,15 @@ static gboolean retrieve_slot_states_via_dbus(GError **error)
 	GBusType bus_type = (!g_strcmp0(g_getenv("DBUS_STARTER_BUS_TYPE"), "session"))
 	                    ? G_BUS_TYPE_SESSION : G_BUS_TYPE_SYSTEM;
 	GError *ierror = NULL;
-	RInstaller *proxy;
+	g_autoptr(RInstallerProxy) proxy = NULL;
```

Comment:
It seems strange to introduce casts in both directions just to get g_autoptr working. The final patch in https://bugzilla.gnome.org/show_bug.cgi?id=763379 (merged in glib 2.50) adds cleanup support for the main interface (RInstaller).
So I'd prefer to fix the leak with a normal g_object_unref/g_clear_object and add a note to switch to g_autoptr with glib 2.50.